### PR TITLE
[ConstraintGraph] NFC: Remove flags from inference methods

### PR DESCRIPTION
**Pull request by xedin**
- Add a `notifyReferencedVars` method to help propagate information to type variables referenced by current one.
- Remove `notifyReferencedVars` flag from all inference related methods and use `notifyReferencedVars` where
   appropriate.

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test Windows platform

**Comment by xedin** [comment]
@swift-ci please smoke test Windows platform

File: lib/Sema/ConstraintGraph.cpp [hunk 1]
```diff
@@ -289,52 +311,51 @@ static bool isUsefulForReferencedVars(Constraint *constraint) {
...
     if (!fixedType)
       getCurrentBindings().retract(constraint);
   } else {
     auto *repr =
         getTypeVariable()->getImpl().getRepresentative(/*record=*/nullptr);
-    CG[repr].retractFromInference(constraint, /*notifyReferencedVars=*/false);
+    CG[repr].retractFromInference(constraint);
   }
 
+  /*
```
Comment (LucianoPAlmeida) [inline lib/Sema/ConstraintGraph.cpp:346]: Hey @xedin :)
We end up with this commented, is that correct?

Comment (xedin) [reply lib/Sema/ConstraintGraph.cpp:346]: Apparently I forgot to push a commit that cleans that up...

Comment (xedin) [reply lib/Sema/ConstraintGraph.cpp:346]: https://github.com/apple/swift/pull/37841

**Pull request merged**